dataset/src/main/scala/frameless/macros/ColumnMacros.scala

```diff
@@ -0,0 +1,377 @@
+package frameless.macros
+
+import frameless.{TypedColumn, TypedEncoder}
+import org.apache.spark.sql.ColumnName
+
+import scala.collection.immutable.Queue
+import scala.reflect.macros.{TypecheckException, whitebox}
+
+class ColumnMacros(val c: whitebox.Context) {
+  import c.universe._
+
+  private val TypedExpressionEncoder = reify(frameless.TypedExpressionEncoder)
+  private val TypedDataset = reify(frameless.TypedDataset)
+  private val ColumnName = weakTypeOf[ColumnName]
+
+  private def toColumn[A : WeakTypeTag, B : WeakTypeTag](
+    selectorStr: String,
+    encoder: c.Expr[TypedEncoder[B]]
+  ): Tree = {
+
+    val A = weakTypeOf[A].dealias
+    val B = weakTypeOf[B].dealias
+
+    val typedCol = appliedType(
+      weakTypeOf[TypedColumn[_, _]].typeConstructor, A, B
+    )
+
+
+    val datasetCol = c.typecheck(
+      q"new $ColumnName($selectorStr).as[$B]($TypedExpressionEncoder.apply[$B]($encoder))"
```

Comment:
@jeremyrsmith is it possible to get rid of the "ds()" requirement all together? 

```scala
ds.groupBy(_.a).aggExpr(x => (sum(x.b)), count(x.b)) ).selectExpr(x => (x._1, x._2 / x._3))
```

It will make the syntax a bit more consistent. Maybe it's a lot of work? what do you think?